File: Data Structures & Algorithms/climbing-stairs/submission-0.py
class Solution:
    def climbStairs(self, n: int) -> int:
        self.res = 0
        def dfs(num: int):
            if num == 0:
                self.res += 1
            elif num < 0:
                return
            dfs(num - 1)
            dfs(num - 2)
        dfs(n)
        return self.res
            

File: Data Structures & Algorithms/palindromic-substrings/submission-0.cpp
class Solution {
public:
    int countSubstrings(string s) {
        int res = 0;

        for (int i = 0; i < s.size(); i++){
            // odd len
            int leftPtr = i;
            int rightPtr = i;
            while(leftPtr >= 0 && rightPtr < s.size() && s[leftPtr] == s[rightPtr]){
                res += 1;
                leftPtr -= 1;
                rightPtr += 1;
            }
            
            // even len
            leftPtr = i;
            rightPtr = i + 1;
            while(leftPtr >= 0 && rightPtr < s.size() && s[leftPtr] == s[rightPtr]){
                res += 1;
                leftPtr -= 1;
                rightPtr += 1;
            }
        }
        return res;
    }
};
